op-sync-tester/synctester/backend/backend.go

```diff
@@ -0,0 +1,117 @@
+package backend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/locks"
+	"github.com/ethereum-optimism/optimism/op-sync-tester/metrics"
+	"github.com/ethereum-optimism/optimism/op-sync-tester/synctester/backend/config"
+	"github.com/ethereum-optimism/optimism/op-sync-tester/synctester/frontend"
+
+	sttypes "github.com/ethereum-optimism/optimism/op-sync-tester/synctester/backend/types"
+)
+
+type sessionKeyType struct{}
+
+var CtxKeySession = sessionKeyType{}
```

Comment:
Go nit: it's best to keep the context-key private, and add two public methods instead, to read from the context, and wrap the context with a new value. That way arbitrary external users cannot add unexpected types of values with the same key, giving better type guarantees.